jar-submission-test/src/main/java/com/hazelcast/jet/tests/jarsubmission/JarSubmissionTest.java

```diff
@@ -0,0 +1,310 @@
+            if (exception != null) {
+                throw exception;
+            }
+        }
+    }
+
+    @Override
+    protected void teardown(Throwable t) throws Exception {
+    }
+
+    protected void testBatchSmallJar(HazelcastInstance client) throws IOException {
+        testBatchJar(client, smallJarPath, "JarSubmissionTestBatchSmall", pauseBetweenSmallJarJobs,
+                SMALL_JAR_LOG_JOB_COUNT_THRESHOLD);
+    }
+
+    protected void testBatchLargeJar(HazelcastInstance client) throws IOException {
+        testBatchJar(client, largeJarPath, "JarSubmissionTestBatchLarge", pauseBetweenLargeJarJobs,
+                LARGE_JAR_LOG_JOB_COUNT_THRESHOLD);
+    }
+
+    protected void testStreamSmallJar(HazelcastInstance client) throws IOException {
+        testStreamJar(client, smallJarPath, "JarSubmissionTestStreamSmall", pauseBetweenSmallJarJobs,
+                SMALL_JAR_LOG_JOB_COUNT_THRESHOLD);
+    }
+
+    protected void testStreamLargeJar(HazelcastInstance client) throws IOException {
+        testStreamJar(client, largeJarPath, "JarSubmissionTestStreamLarge", pauseBetweenLargeJarJobs,
+                LARGE_JAR_LOG_JOB_COUNT_THRESHOLD);
+    }
+
+    protected void testBatchJar(HazelcastInstance client, String jarPath, String prefix, int sleep, int logThreshold) {
+        Path jarFile = Paths.get(jarPath);
+        assertTrue("testing file for " + prefix + " does not exist", Files.exists(jarFile));
+
+        JetService jet = client.getJet();
+
+        long begin = System.currentTimeMillis();
+        long jobCount = 0;
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            ArrayList<String> params = new ArrayList<>();
+            params.add(Long.toString(jobCount));
+            params.add(prefix);
+            // true means batch job
+            params.add("true");
+            SubmitJobParameters jobParams = new SubmitJobParameters()
+                    .setJobParameters(params)
+                    .setJobName(prefix + jobCount)
+                    .setJarPath(jarFile);
+            // set non-existing main class for 10% of jobs
+            if (jobCount % 10 == 5) {
+                jobParams.setMainClass("not.existing.Main");
+            }
+
+            try {
+                jet.submitJobFromJar(jobParams);
+            } catch (Exception ex) {
+                if (jobCount % 10 == 5) {
+                    // expected
+                } else {
+                    throw ex;
+                }
+            }
+
+            if (jobCount % 10 != 5) {
+                Job job = getJobEventually(jet, prefix + jobCount);
+                Exception jobException = null;
+                try {
+                    job.join();
+                } catch (Exception ex) {
+                    jobException = ex;
+                }
+
+                // check result
+                IMap<String, String> map = client.getMap(prefix);
+                if (jobCount % 10 == 0) {
```

Comment:
Why this condition?

Reply:
It's from the pipeline - https://github.com/hazelcast/hazelcast-jet-ansible-tests/pull/168/files#diff-365662a530d727fe81515c1f1ba39ea3dba27d5d214be3a6bf3edd177c8bec7dR36. We want to pipeline to fail sometimes (in 10% of executions) to check whether there's not any memory leak in this case.